packages/theming-core/src/generics/themeRegistry.test.ts

```diff
@@ -0,0 +1,125 @@
+import { createThemeRegistry, IThemeRegistry } from './themeRegistry';
+
+interface IPartialTestTheme {
+  val: number;
+}
+
+interface ITestTheme {
+  val: number;
+  sum: number;
+}
+
+function _resolveTestTheme(definition: IPartialTestTheme | undefined, parent: ITestTheme): ITestTheme {
+  const val: number = (definition && definition.val) || 0;
+  return {
+    val,
+    sum: parent.sum + val
+  };
+}
+
+function _addTen(theme: ITestTheme): IPartialTestTheme {
+  return {
+    val: theme.val + 10
+  };
+}
+
+function _testRegistry(): IThemeRegistry<ITestTheme, IPartialTestTheme> {
+  return createThemeRegistry<ITestTheme, IPartialTestTheme>({ val: 0, sum: 0 }, _resolveTestTheme);
+}
+
+describe('theme registry tests', () => {
+  it('immediate query of default', () => {
+    const registry = _testRegistry();
+    const def = registry.getTheme();
+    expect(def.sum).toEqual(0);
+  });
+
+  it('default overriding from platform', () => {
+    const registry = _testRegistry();
```

Comment:
to reduce repetition, you can put `let registry;` in the describe scope, and add a `beforeEach` to initialize:

```tsx
beforeEach(() => {
  registry = _testRegistry();
});
```